src/folding/circuits/nimfs_verifier.rs

```diff
@@ -1,2 +1,78 @@
 // hypernova nimfs verifier circuit
 // see section 5 in https://eprint.iacr.org/2023/573.pdf
```

Comment:
I agree with having the `EqEvalGadget` in `src/folding/circuits`, since it will be reused also by some versions of ProtoGalaxy, but this directory should be agnostic to hypernova (thus not containing references to HyperNova's NIMFS).
How would you see to change the file name into something like `src/folding/circuits/utils.rs` so it can contain other utilities in the future, but the future `nimfs` specific gadgets would be at `src/folding/hypernova/circuits.rs`?

Reply:
Yes definitely!